core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2207,20 +2207,22 @@ class SparkContext(config: SparkConf) extends Logging {
    * Cancel a given job if it's scheduled or running.
    *
    * @param jobId the job ID to cancel
+   * @param reason optional reason for cancellation
    * @note Throws `InterruptedException` if the cancel message cannot be sent
    */
-  def cancelJob(jobId: Int) {
-    dagScheduler.cancelJob(jobId)
+  def cancelJob(jobId: Int, reason: String = "") {
```

Comment:
actually you need to define a new function with two parameters and keep the old one to maintain binary compatibility.